src/main/java/org/opensearch/security/tools/democonfig/Installer.java

```diff
@@ -82,60 +142,45 @@ static void printScriptHeaders() {
 
     /**
      * Reads the options passed to the script
-     * @param options an array of strings containing options passed to the script
+     * @param args an array of strings containing options passed to the script
      */
-    static void readOptions(String[] options) {
+    void readOptions(String[] args) {
         // set script execution dir
-        SCRIPT_DIR = options[0];
-
-        for (int i = 1; i < options.length; i++) {
-            switch (options[i]) {
-                case "-y":
-                    assumeyes = true;
-                    break;
-                case "-i":
-                    initsecurity = true;
-                    break;
-                case "-c":
-                    cluster_mode = true;
-                    break;
-                case "-s":
-                    skip_updates = 0;
-                    break;
-                case "-t":
-                    environment = ExecutionEnvironment.TEST;
-                    break;
-                case "-h":
-                case "-?":
-                    showHelp();
-                    return;
-                default:
-                    System.out.println("Invalid option: " + options[i]);
+        SCRIPT_DIR = args[0];
+
+        CommandLineParser parser = new DefaultParser();
+        try {
+            CommandLine line = parser.parse(options, args);
+
+            if (line.hasOption("h")) {
+                showHelp();
+                return;
             }
+            assumeyes = line.hasOption("y");
+            initsecurity = line.hasOption("i");
+            cluster_mode = line.hasOption("c");
+            skip_updates = line.hasOption("s") ? 0 : -1;
```

Comment:
Why is this the only boolean redefined into a number?

Reply:
this is to defined exit code used later on line 196 in SecuritySettingsConfigurer.java